.yarn/patches/playroom-npm-0.39.0-24448abd08.patch

```diff
@@ -0,0 +1,890 @@
++                      role="presentation"
++                      className={styles.regenerateIcon}
++                    >
++                      <path
++                        fill="currentColor"
++                        d="M18.2 7.905h-4.405v-1.25h2.321a6.75 6.75 0 0 0-10.865 5.45H4l-.001-.1A8 8 0 0 1 16.95 5.72V3.5h1.25zM5.8 16.105h4.405v1.25H7.884a6.75 6.75 0 0 0 10.866-5.45H20v.1A8 8 0 0 1 7.05 18.29v2.22H5.8z"
++                      />
++                    </svg>
++                  </button>
++                </div>
++              </div>
++            ) : (
++              <div
++                key={idx}
++                className={classNames(styles.userBubble, {
++                  [styles.disabledMessage]:
++                    regeneratingIndex !== null && regeneratingIndex < idx,
++                })}
++              >
++                {message.content.type === 'text' ? (
++                  message.content.text
++                ) : (
++                  <img src={message.content.url} width="100%" />
++                )}
++              </div>
++            )
++          )}
++          {isWaitingForResponse && (
++            <div className={styles.loadingBubble}>
++              <AnnimatedDots />
++            </div>
++          )}
++          {error && <div className={styles.errorBubble}>{error}</div>}
++        </div>
++        <textarea
++          ref={textAreaRef}
++          onKeyUp={(e) => {
++            if (e.key === 'Enter' && !e.shiftKey) {
++              sendMessage({ type: 'text', text: e.currentTarget.value.trim() });
++              e.currentTarget.value = '';
++            }
++          }}
++          disabled={isLoading || Boolean(error)}
++          className={styles.textarea}
++          style={{
++            borderColor: error ? `red` : undefined,
++          }}
++          placeholder={
++            isLoading
++              ? 'Generating...'
++              : error
++              ? 'There was an error'
++              : aiConversation.conversationId
++              ? 'Ask for changes'
++              : 'What do you want to build?'
++          }
++        />
++      </div>
++    </ToolbarPanel>
++  );
++};
+diff --git a/src/Playroom/Toolbar/Toolbar.tsx b/src/Playroom/Toolbar/Toolbar.tsx
+index 389401bc0df0a8cba17cc0f4a53229027bdc81f2..9b2fd4f0332ea320013f33009ba07b30855d5744 100644
+--- a/src/Playroom/Toolbar/Toolbar.tsx
++++ b/src/Playroom/Toolbar/Toolbar.tsx
+@@ -18,6 +18,7 @@ import SettingsIcon from '../icons/SettingsIcon';
+ import { isMac } from '../../utils/formatting';
+ 
+ import { CSSTransition } from 'react-transition-group';
++import AiPanel, { aiBaseUrl } from '../AiPanel/AiPanel';
+ 
+ interface Props {
+   themes: PlayroomProps['themes'];
+@@ -27,7 +28,21 @@ interface Props {
+ 
+ const ANIMATION_TIMEOUT = 300;
+ 
++const useWithAiEnabled = () => {
++  const [canUseAi, setCanUseAi] = useState(false);
++  useEffect(() => {
++    fetch(`${aiBaseUrl}/health-check`).then((response) => {
```

Comment:
only show the AI assistant panel when the backend service can be reached (for the moment this will only happen inside the VPN)